File: 프로그래머스/1/42862. 체육복/체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var students = Array(repeating: 1, count: n)
    for i in lost { students[i - 1] -= 1 }
    for i in reserve { students[i - 1] += 1 }
    for i in 0..<n {
        if students[i] <= 0 {
            if i > 0 && students[i - 1] > 1 {
                students[i - 1] -= 1
                students[i] += 1
            }
            else if i < n - 1 && students[i + 1] > 1 {
                students[i + 1] -= 1
                students[i] += 1
            } 
        }
    }
    return students.filter { $0 >= 1 }.count
}

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> st(n+1, 1);
    for(int l : lost) st[l]--;
    for(int r : reserve) st[r]++;
    for(int i = 1; i <= n; i++) {
        if(st[i] == 0) {
            if(st[i-1] == 2) {
                st[i-1]--; st[i]++;
            }
            else if(st[i+1] == 2) {
                st[i+1]--; st[i]++;
            }
        }
    }
    for(int i = 1; i <= n; i++) {
        if(st[i] >= 1) answer++;
    }
    return answer;
}